src/components/uui-pagination/uui-pagination.element.ts

```diff
@@ -0,0 +1,306 @@
+    this.current--;
+    this.dispatchEvent(new UUIPaginationEvent(UUIPaginationEvent.CHANGE));
+  }
+
+  protected goToPage(page: number) {
+    this.current = page;
+    this.dispatchEvent(new UUIPaginationEvent(UUIPaginationEvent.CHANGE));
+  }
+
+  /** When having limited display of page-buttons and clicking a page-button that changes the current range, the focus stays on the position of the clicked button which is not anymore representing the number clicked, therefor we move focus to the button that represents the current page. */
+  protected setFocusActivePageButton() {
+    requestAnimationFrame(() => {
+      // for none range changing clicks we need to ensure a rendering before querying.
+      const activeButtonElement = this.renderRoot.querySelector<HTMLElement>(
+        '.active-button'
+      );
+      if (activeButtonElement) {
+        activeButtonElement.focus();
+      }
+    });
+  }
+
+  protected firstButtonTemplate() {
+    return html`<uui-button
+      compact
+      look="outline"
+      class="nav-button"
+      role="listitem"
+      aria-label="Go to first page"
+      .disabled=${1 === this._current}
+      @click=${() => this.goToPage(1)}
+    >
+      First
+    </uui-button>`;
+  }
+
+  protected previousButtonTemplate() {
+    return html`<uui-button
+      compact
+      look="outline"
+      class="nav-button"
+      role="listitem"
+      aria-label="Go to previous page"
+      .disabled=${this.current === 1}
+      @click=${this.goToPreviousPage}
+    >
+      Previous
+    </uui-button>`;
+  }
+
+  protected nextButtonTemplate() {
+    return html`<uui-button
+      compact
+      look="outline"
+      role="listitem"
+      class="nav-button"
+      aria-label="Go to next page"
+      .disabled=${this.current === this.total}
+      @click=${this.goToNextPage}
+    >
+      Next
+    </uui-button>`;
+  }
+
+  protected lastButtonTemplate() {
+    return html`
+      <uui-button
+        compact
+        look="outline"
+        role="listitem"
+        class="nav-button"
+        aria-label="Go to last page"
+        ?disabled=${this.total === this._current}
+        @click=${() => this.goToPage(this.total)}
+      >
+        Last
+      </uui-button>
+    `;
+  }
+
+  protected dotsTemplate() {
+    return html`<uui-button
+      compact
+      look="outline"
+      tabindex="-1"
+      class="dots-button"
+      >...</uui-button
+    > `;
+  }
+
+  protected pageTemplate(page: number) {
+    return html`<uui-button
+      compact
+      look=${page === this._current ? 'primary' : 'outline'}
+      role="listitem"
+      aria-label="Go to page ${page}"
+      class=${page === this._current ? 'active-button' : ''}
+      tabindex=${page === this._current ? '-1' : ''}
+      @click=${() => {
+        if (page === this._current) return;
+        this.goToPage(page);
+        this.setFocusActivePageButton();
+      }}
+      >${page}</uui-button
+    >`;
+  }
+
+  protected navigationLeftTemplate() {
```

Comment:
Now we always show first, next, previous and last buttons. aka. i removed the optional rendering for these.